### PR TITLE
[Fix] Adjust the minimum supported python version to 3.7

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python
```
Comment (ZwwWayne) [inline setup.py:1]: this line seems to be an unnecessary modification

File: setup.py [hunk 2]
```diff
@@ -23,13 +25,7 @@ def readme():
 def get_version():
     with open(version_file, 'r') as f:
         exec(compile(f.read(), version_file, 'exec'))
-    import sys
-
-    # return short version for sdist
-    if 'sdist' in sys.argv or 'bdist_wheel' in sys.argv:
```
Comment (ZwwWayne) [inline setup.py:29]: these code might be necessary, we need to check whether the delete is safe.

**Comment by JingweiZhang12** [comment]
Please pull the latest code, which has fixed the UT.

File: setup.py (hunk 2 above)
Comment (Xiangxu-0103) [reply setup.py:29]: I have checked that the delete is safe.

**Pull request merged**